daemon/state.go

```diff
@@ -144,21 +144,25 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
 
 		state.restored = append(state.restored, ep)
 
-		delete(existingEndpoints, ep.IPv4.String())
-		delete(existingEndpoints, ep.IPv6.String())
+		if existingEndpoints != nil {
+			delete(existingEndpoints, ep.IPv4.String())
```

Comment:
This should be safe on a nil map (I just tested https://play.golang.org/p/rr9va0dF2kp), but perhaps you wanted to make it clear that it did nothing when nil?

Reply:
Yeah, it's more of a documentation purpose to make it clear that existingEndpoints is expected to be nil in some cases.